src/Text/LLVM/AST.hs

```diff
@@ -368,22 +418,31 @@ parseDataLayout str =
       do c <- letter
          case c of
            'e' -> return ElfMangling
+           'l' -> return GOFFMangling
            'm' -> return MipsMangling
            'o' -> return MachOMangling
            'w' -> return WindowsCoffMangling
+           'x' -> return WindowsX86CoffMangling
+           'a' -> return XCOFFMangling
            _   -> mzero
 
+    pAlignment :: Parser Alignment
+    pAlignment = Alignment <$> pInt <*> pCOInt
+
+    pStorage :: Parser Storage
+    pStorage = Storage <$> pInt <* char ':' <*> pAlignment
+
     pInt :: Parser Int
     pInt = read <$> many1 digit
 
-    pInt0 :: Parser Int
-    pInt0 = pInt <|> return 0
+    pInt' :: Char -> Parser Int
+    pInt' d0 = read . (d0:) <$> many digit
```

Comment:
`pInt'` is a bit hard to distinguish from `pInt`, despite having pretty different type signatures. Perhaps this should be renamed to something like `pIntPrefix` or `pIntLookahead`?